File: Data Structures & Algorithms/two-integer-sum/submission-8.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        diff = {}
        remaining_diff = target
        for i in range(len(nums)):
            if (target - nums[i]) in diff:
                return [diff[target-nums[i]], i]
            else:
                diff[nums[i]] = i
            

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-8.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int window = 0; int l = 0; int res = INT_MAX;

        for (int r = 0; r < nums.size(); ++r) {
            window += nums[r];

            while (window >= target) {
                res = min(res, r-l+1);
                window -= nums[l];
                l++;
            }
        }

        if (res == INT_MAX) {
            return 0;
        } return res;
    }
};